arangod/Scheduler/Scheduler.cpp

```diff
@@ -208,24 +208,45 @@ Scheduler::~Scheduler() {
 // do not pass callback by reference, might get deleted before execution
 void Scheduler::post(std::function<void()> const callback) {
   // increment number of queued and guard against exceptions
+  //  (this incQueued() manipulates the atomic _counters in a sequentially-consistent
+  //   manner.  isStopping() uses same atomic _counters)
   incQueued();
 
-  auto guardQueue = scopeGuard([this]() { decQueued(); });
+  // implies if _ioContext still valid (defense against shutdown races)
+  if (!isStopping()) {
+
+    auto guardQueue = scopeGuard([this]() { decQueued(); });
+
+    // capture without self, ioContext will not live longer than scheduler
+    _ioContext->post([this, callback]() {
+        // start working
+        JobGuard jobGuard(this);
+        jobGuard.work();
+
+        // reduce number of queued now
+        decQueued();
 
-  // capture without self, ioContext will not live longer than scheduler
-  _ioContext->post([this, callback]() {
-    // start working
+        callback();
+      });
+
+    // no exception happened, cancel guard
+    guardQueue.cancel();
+  } else {
+    // increase number of working (must precede decQueue() to keep shutdown looping)
     JobGuard jobGuard(this);
     jobGuard.work();
 
     // reduce number of queued now
     decQueued();
 
+    // this post is coming late in application shutdown,
+    //  might be essential ... process in-line after heavy complaining
+    LOG_TOPIC(WARN, Logger::THREADS)
+      << "Scheduler::post() called after isStopping() set.";
+    TRI_ASSERT(false);
+    std::abort();   /// temporary ... debugging within Jenkins
```

Comment:
@matthewvon You forgot removing the `abort` here.

Reply:
Corrected, thank you.